lib/ansible/modules/identity/opendj/opendj_backend.py

```diff
@@ -0,0 +1,349 @@
+   - This module will create/delete a backend in OpenDJ.
+   - This module will create a 'root' entry, this is done via temporary creating a file in /tmp and when import is done, it is deleted.
+version_added: "2.2"
+author:
+    - Werner Dijkerman
+options:
+    opendj_bindir:
+        description:
+            - The path to the bin directory of OpenDJ.
+        required: false
+        default: /opt/opendj/bin
+    hostname:
+        description:
+            - The hostname of the OpenDJ server.
+        required: true
+    port:
+        description:
+            - The Admin port on which OpenDJ is listening.
+        required: true
+    password:
+        description:
+            - The password for the cn=Directory Manager user.
+            - Either password or passwordfile is needed.
+        required: false
+    passwordfile:
+        description:
+            - Location to the password file which holds the password for user configured in bind_dn.
+            - Either password or passwordfile is needed.
+        required: false
+    name:
+        description:
+            - The name of the backend to create.
+        required: true
+    bind_dn:
+        description:
+            - The username to connect to OpenDJ.
+        required: false
+        default: cn=Directory Manager
+    base_dn:
+        description:
+            - The base DN. Example dc=example,dc=com.
+        required: True
+    type:
+        description:
+            - The type of backend you want to use for OpenDJ.
+        required: false
+        choices: ["je", "pdb"]
+        default: "je"
+    enabled:
+        description:
+            - If the backend needs to be enabled when the backend is created.
+        required: false
+        choices: ['True', 'False']
+        default: "True"
+    state:
+        description:
+            - If configuration needs to be added/deleted
+        required: false
+        choices: ['present', 'absent']
+        default: "present"
+'''
+
+EXAMPLES = '''
+    - name: "Create Pizza Backend"
+      action: opendj_backend
+              hostname=localhost
+              port=4444
+              password=password
+              name=myPizzaBackend
+              bind_dn="cn=Directory Manager"
+              base_dn=dc=ansible,dc=example,dc=com
+              enabled=true
+
+    - name: "Delete Pizza Backend"
+      action: opendj_backend
+              hostname=localhost
+              port=4444
+              password=password
+              name=myPizzaBackend
+              bind_dn="cn=Directory Manager"
+              base_dn=dc=ansible,dc=example,dc=com
+              type=pdb
+              enabled=true
+              state=absent
+
+'''
+
+RETURN = '''
+
+'''
+
+import subprocess
+import time
+from tempfile import mkstemp
+import os
+import re
+
+
+class Backend(object):
+    def __init__(self, module):
+        self._module = module
+
+    def get_backend(self, opendj_bindir, hostname, port, password_method, bind_dn):
```

Comment:
Can you use instance variable for this, so that they will be available for other methods in class Backend?